site/helm-integration.md

```diff
@@ -1,119 +1,195 @@
-# Design overview
+---
+title: Using Flux with Helm
+menu_order: 90
+---
+
+# Using Flux with Helm
+
+You can release charts to your cluster via "GitOps", by combining Flux
+and the Flux Helm Operator (also in
+[weaveworks/flux](https://github.com/weaveworks/flux)).
+
+The essential mechanism is this: the declaration of a Helm release is
+represented by a custom resource, specifying the chart and its
+values. If you put such a resource in your git repo as a file, Flux
+will apply it to the cluster, and once it's in the cluster, the Helm
+Operator will make sure the release exists by installing or upgrading
+it.
+
+## The `FluxHelmRelease` custom resource
+
+Each release of a chart is declared by a `FluxHelmRelease`
+resource. The schema for these resources is given in [the custom
+resource definition](../deploy-helm/flux-helm-release-crd.yaml). They
+look like this:
+
+```yaml
+---
+apiVersion: flux.weave.works/v1beta1
+kind: FluxHelmRelease
+metadata:
+  name: rabitmq
+  namespace: default
+spec:
+  releaseName: rabbitmq
+  chart:
+    repository: https://kubernetes-charts.storage.googleapis.com/
+    name: rabbitmq
+    version: 3.3.6
+  values:
+    replicas: 1
+```
+
+The `releaseName` will be given to Helm as the release name. If not
+supplied, it will be generated by affixing the namespace to the
+resource name. In the above example, if `releaseName` were not given,
+it would be generated as `default-rabbitmq`. Because of the way Helm
+works, release names must be unique in the cluster.
+
+The `chart` section gives a pointer to the chart; in this case, to a
+chart in a Helm repo. Since the helm operator is running in your
+cluster, and doesn't have access to local configuration, the
+repository is given as a URL rather than an alias (the URL in the
+example is what's usually aliased as `stable`). The `name` and
+`version` specify the chart to release.
+
+The `values` section is where you provide the value overrides for the
+chart. This is as you would put in a `values.yaml` file, but inlined
+into the structure of the resource.
+
+### Using a chart from a Git repo instead of a Helm repo
+
+You can refer to a chart from a _git_ repo, rather than a chart repo,
+with a `chart:` section like this:
+
+```yaml
+spec:
```

Comment:
I'm a bit confused with the resource context here, does this refer to a `FluxHelmRelease`, in other words:

```yaml
---
apiVersion: helm.integrations.flux.weave.works/v1alpha2
kind: FluxHelmRelease
metadata:
  name: flux
  namespace: flux
  labels:
    chart: flux
spec:
  chartGitPath: flux
  releaseName: flux
```

```yaml
---
apiVersion: helm.integrations.flux.weave.works/v1alpha2
kind: FluxHelmRelease
metadata:
  name: flux
  namespace: flux
  labels:
    chart: flux
spec:
  chart:
    git: git@github.com:weaveworks/flux
    ref: master
    path: charts/flux
  releaseName: flux
```

Reply:
@squaremo 
What is the correct version . `flux.weave.works/v1beta1` or `helm.integrations.flux.weave.works/v1alpha2` ?

Reply:
For the helm operator from this branch, `flux.weave.works/v1beta1`. The difference is mainly the option of supplying a chart repo/name/version or a git URL/branch.

Reply:
thanks. I'm hoping to try new master soon for this functionality `https://github.com/weaveworks/flux/pull/1450` but not entirely sure if i need what inside this open PR.

Reply:
@stephenmoloney you don't need anything from this PR to run the deps update